libnetwork/network.go

```diff
@@ -1524,9 +1524,9 @@ func (n *Network) ipamAllocate() error {
 	return err
 }
 
-func (n *Network) requestPoolHelper(ipam ipamapi.Ipam, addressSpace, preferredPool, subPool string, options map[string]string, v6 bool) (string, *net.IPNet, map[string]string, error) {
+func (n *Network) requestPoolHelper(ipam ipamapi.Ipam, addressSpace, preferredPool, subPool string, options map[string]string, v6 bool) (poolID string, pool *net.IPNet, meta map[string]string, err error) {
```

Comment:
Oh, now I realize your initial choice of `preferredPool` is coming from here. I think we should rename `preferredPool` and `subPool` here too, to convey there're hard requirements and not just preference.